### ISSUE TITLE
Ready to use Docker container/dockerfile for GEOSX

### ISSUE THREAD

**Issue by pradhyumna85** [GEOS-DEV/GEOS#2788]
**What is the requested feature?**
A ready-to-use container image for CPU/GPU, enabling quick start use of GEOSX.

**Is your request related to a specific problem?**
It's kind of tedious for someone new (For example, a reservoir engineer) to build/compile GEOSX from the source.

**Describe the solution you'd like**
A Dockerfile and container images for CPU/GPU builds.

**Describe alternatives you've considered**
For now, the only option seems to be to build from the source.

**Comment by TotoGaz** [comment]
Hello, did you consider https://geosx-geosx.readthedocs-hosted.com/en/latest/docs/sphinx/developerGuide/Contributing/UsingDocker.html and https://geosx-geosx.readthedocs-hosted.com/en/latest/docs/sphinx/developerGuide/Contributing/InstallWin.html?

These pages are tagged `[Unsupported]` as we do not want to commit tightly to this solution and we still want to be free of any modification. But the proper tag would better be `[Best effort]`...

**Comment by pradhyumna85** [comment]
Hi @TotoGaz, yeah for me it's not too difficult to build GEOSX in docker. My thought was that a ready-to-go simple cross-platform docker container solution would help new/newbie users start quickly.

I have created a kind of generic Dockerfile and cmake file to build a container with GEOSX for CPU: https://github.com/pradhyumna85/build-geosx-image/tree/generic-docker-config

Let me know if we can or have plans to add something similar to the GEOS repo, I am happy to contribute.

Thanks

**Comment by TotoGaz** [comment]
Hello. We're considering offering solutions like the one you describe.
But there are some drawbacks, since we want to be sure that such a pre-built version would not be used to assess the performances of `geos` (you should build `geos` for your specific configuration for that).
For the GPU, there are questions about the compute capabilities offered by CUDA and that `geos` should be compiled against.

Nevertheless, I think we're considering working on this as a mid term solution, but nothing is planned so far.

**Comment by pradhyumna85** [comment]
Got it @TotoGaz.
What about having 2 Dockerfiles + cmake templates, 1 each for CPU and GPU(cuda) which would give a very quick boilerplate.
Similar approach could be found for other projects:
- https://github.com/KoboldAI/KoboldAI-Client (see the "docker-*" directories)
- https://github.com/abetlen/llama-cpp-python (see the "docker" directory)
- https://github.com/getumbrel/llama-gpt (see the different docker compose files)


For GPU(cuda) in case of GEOSX, we can take the compute capability also as a build ARG in the Dockerfile.

Basically by adding the relevant build arguments in Dockerfile and passing them in docker build and Done!

**Comment by paveltomin** [comment]
https://github.com/GEOS-DEV/GEOS/issues/1619
